core/src/test/scala/unit/kafka/admin/DeleteTopicTest.scala

```diff
@@ -20,23 +20,37 @@ import java.util
 import java.util.concurrent.ExecutionException
 import java.util.{Collections, Optional, Properties}
 
-import scala.collection.Seq
-import kafka.log.UnifiedLog
-import kafka.zk.TopicPartitionZNode
-import kafka.utils.TestUtils
-import kafka.server.{KafkaConfig, KafkaServer, QuorumTestHarness}
-import org.junit.jupiter.api.Assertions._
-import org.junit.jupiter.api.{AfterEach, Test}
 import kafka.common.TopicAlreadyMarkedForDeletionException
 import kafka.controller.{OfflineReplica, PartitionAndReplica, ReplicaAssignment, ReplicaDeletionSuccessful}
+import kafka.log.UnifiedLog
+import kafka.server
+import kafka.server.{KafkaBroker, KafkaConfig, KafkaServer, QuorumTestHarness}
+import kafka.utils.Implicits.PropertiesOps
+import kafka.utils.{TestInfoUtils, TestUtils}
+import kafka.zk.TopicPartitionZNode
 import org.apache.kafka.clients.admin.{Admin, AdminClientConfig, NewPartitionReassignment, NewPartitions}
-import org.apache.kafka.common.TopicPartition
+import org.apache.kafka.common.{TopicPartition, TopicPartitionInfo}
 import org.apache.kafka.common.errors.UnknownTopicOrPartitionException
+import org.apache.kafka.common.network.{ListenerName, Mode}
+import org.apache.kafka.common.security.auth.SecurityProtocol
+import org.junit.jupiter.api.Assertions.{assertTrue, _}
+import org.junit.jupiter.api.function.Executable
+import org.junit.jupiter.api.{AfterEach, TestInfo}
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.ValueSource
+import org.opentest4j.AssertionFailedError
+
+import scala.collection.{Map, Seq}
 import scala.jdk.CollectionConverters._
 
 class DeleteTopicTest extends QuorumTestHarness {
 
-  var servers: Seq[KafkaServer] = Seq()
+  var brokers: Seq[KafkaBroker] = Seq()
```

Comment:
Looks like you want to at least use `KafkaServerTestHarness`. Take a look at `IntegrationTestHarness` too.

Reply:
I thought it would be better to bring up brokers manually for each test since many require different sets of configs (KRaft doesn't yet support dynamic reconfiguration) and different number of brokers (we _could_ just always bring up the max required).